packages/builder/src/pages/builder/portal/manage/users/_components/ImportUsersModal.svelte

```diff
@@ -0,0 +1,157 @@
+<script>
+  import {
+    Body,
+    ModalContent,
+    RadioGroup,
+    Multiselect,
+    notifications,
+  } from "@budibase/bbui"
+  import { groups, auth } from "stores/portal"
+  import { emailValidator } from "../../../../../../helpers/validation"
+  import { Constants } from "@budibase/frontend-core"
+
+  const BYTES_IN_MB = 1000000
+  const FILE_SIZE_LIMIT = BYTES_IN_MB * 5
+  const MAX_USERS_UPLOAD_LIMIT = 1000
+  export let createUsersFromCsv
+
+  let files = []
+  let csvString = undefined
+  let userEmails = []
+  let userGroups = []
+  let usersRole = null
+
+  $: invalidEmails = []
+  $: hasGroupsLicense = $auth.user?.license.features.includes(
+    Constants.Features.USER_GROUPS
+  )
+
+  const validEmails = userEmails => {
+    if (userEmails.length > MAX_USERS_UPLOAD_LIMIT) {
+      notifications.error(
+        `Max limit for upload is 1000 users. Please reduce file size and try again.`
```

Comment:
@shogunpurple @mike12345567 @Rory-Powell Checking for max length in these locations. this seem ok?

Reply:
It would probably be beneficial to do it on the backend as well so that this can't be circumvented

Reply:
Nevermind, just seeing your second comment below - all good!